spring-integration-smb/src/main/java/org/springframework/integration/smb/outbound/SmbMessageHandler.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.smb.outbound;
+
+import org.springframework.integration.file.remote.handler.FileTransferringMessageHandler;
+import org.springframework.integration.file.remote.session.SessionFactory;
+import org.springframework.integration.file.support.FileExistsMode;
+import org.springframework.integration.smb.session.SmbRemoteFileTemplate;
+import org.springframework.integration.smb.session.SmbSessionFactory;
+
+import jcifs.smb.SmbFile;
+
+/**
+* The SMB specific {@link FileTransferringMessageHandler} extension. Based on the
+* {@link SmbRemoteFileTemplate}.
+*
+* @author Gregory Bragg
+*
+* @since 1.3
+*
+* @see SmbRemoteFileTemplate
+*/
+public class SmbMessageHandler extends FileTransferringMessageHandler<SmbFile> {
+
+	public SmbMessageHandler(SessionFactory<SmbFile> sessionFactory) {
+		this(new SmbRemoteFileTemplate(sessionFactory));
+		SmbSessionFactory smbSessionFactory = (SmbSessionFactory) sessionFactory;
+		smbSessionFactory.setReplaceFile(true);
+	}
+
+	public SmbMessageHandler(SmbRemoteFileTemplate remoteFileTemplate) {
+		super(remoteFileTemplate);
+	}
+
+	public SmbMessageHandler(SmbRemoteFileTemplate remoteFileTemplate, FileExistsMode mode) {
```

Comment:
I think we need to disable `SmbRemoteFileTemplate`-based ctors for time being.
Just for the case to not encourage end-users to use those deprecated properties!

Reply:
Not sure I understand... you don't want anybody to use SmbRemoteFileTemplate? I'm just following the pattern for FTP channel adapter in some respects...

Reply:
That's OK to use it outside of this `SmbMessageHandler`.
The point is that `setReplaceFile(true)` has to be called on the `SmbSessionFactory` and if we expose a `SmbRemoteFileTemplate`-based ctor, that means end-user will have to call that deprecated `setReplaceFile()` manually.
If we don't have this kind of ctors, people will just use an existing one and and won't call a deprecated API, because, yeah, it is deprecated.

Does it make sense?

Reply:
We are extending `FileTransferringMessageHandler` so I have to support all the constructors in it... How am I going to do that unless I change SmbRemoteFileTemplate like I first proposed in the other branch?

Reply:
Why is that?
There is just enough to have a one ctor based on one super or delegating to another super?
We are not forced to implement all the super ctors.